stream/src/main/scala/org/apache/pekko/stream/impl/fusing/CollectWhile.scala

```diff
@@ -45,10 +45,11 @@ private[pekko] final class CollectWhile[In, Out](pf: PartialFunction[In, Out]) e
 
       override final def onPush(): Unit =
         try {
-          pf.applyOrElse(grab(in), NotApplied) match {
-            case NotApplied             => completeStage()
-            case result: Out @unchecked => push(out, result)
-            case _                      => throw new RuntimeException() // won't happen, compiler exhaustiveness check pleaser
+          val result = pf.applyOrElse(grab(in), NotApplied)
```

Comment:
can we duplicate the code comment from Ops.scala below?

Reply:
OK.